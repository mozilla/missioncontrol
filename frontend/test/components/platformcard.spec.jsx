import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PlatformCard from '../../ui/platformcard';

configure({ adapter: new Adapter() });

describe('PlatformCard', () => {
  it('should display PlatformCard correctly', () => {
    const sampleMeasures = [
        {
          lastUpdated: '2018-12-18T07:40:00Z',
          majorVersions: [
            {
              adjustedCount: 3694128,
              adjustedRate: 0.97,
              count: 3694128,
              fieldDuration: 663300,
              rate: 0.97,
              version: '64',
            },
            {
              adjustedCount: 3638659,
              adjustedRate: 1.61,
              count: 66553946,
              fieldDuration: 4248000,
              rate: 1.15,
              version: '63',
            },
            {
              adjustedCount: 2747677,
              adjustedRate: 1.44,
              count: 30466929,
              fieldDuration: 4148100,
              rate: 1.06,
              version: '62',
            },
            {
              adjustedCount: 2761278,
              adjustedRate: 1.3,
              count: 89205500,
              fieldDuration: 6100800,
              rate: 0.93,
              version: '61',
            },
          ],
          name: 'main_crashes',
          versions: [
            {
              adjustedCount: 3694128,
              adjustedRate: 0.97,
              count: 3694128,
              fieldDuration: 663300,
              rate: 0.97,
              version: '64.0',
            },
            {
              adjustedCount: 3694128,
              adjustedRate: 0.97,
              count: 3694128,
              fieldDuration: 663300,
              rate: 0.97,
              version: '64',
            },
            {
              adjustedCount: 3638659,
              adjustedRate: 1.61,
              count: 66553946,
              fieldDuration: 4248000,
              rate: 1.15,
              version: '63',
            },
            {
              adjustedCount: 2747677,
              adjustedRate: 1.44,
              count: 30466929,
              fieldDuration: 4148100,
              rate: 1.06,
              version: '62',
            },
            {
              adjustedCount: 2761278,
              adjustedRate: 1.3,
              count: 89205500,
              fieldDuration: 6100800,
              rate: 0.93,
              version: '61',
            },
          ],
        },
        {
          lastUpdated: '2018-12-18T07:40:00Z',
          majorVersions: [
            {
              adjustedCount: 5836415,
              adjustedRate: 1,
              count: 5836415,
              fieldDuration: 663300,
              rate: 1.53,
              version: '64',
            },
            {
              adjustedCount: 4163445,
              adjustedRate: 1.84,
              count: 68708257,
              fieldDuration: 4248000,
              rate: 1.19,
              version: '63',
            },
            {
              adjustedCount: 2950592,
              adjustedRate: 1.55,
              count: 30932336,
              fieldDuration: 4148100,
              rate: 1.08,
              version: '62',
            },
            {
              adjustedCount: 3616486,
              adjustedRate: 1.7,
              count: 112777330,
              fieldDuration: 6100800,
              rate: 1.17,
              version: '61',
            }],
            name: 'content_crashes',
            versions: [
              {
                adjustedCount: 5836415,
                adjustedRate: 1.53,
                count: 5836415,
                fieldDuration: 663300,
                rate: 1.53,
                version: '64.0',
              },
              {
                adjustedCount: 5836415,
                adjustedRate: 1.53,
                count: 5836415,
                fieldDuration: 663300,
                rate: 1.53,
                version: '64',
              },
              {
                adjustedCount: 4163445,
                adjustedRate: 1.84,
                count: 68708257,
                fieldDuration: 4248000,
                rate: 1.19,
                version: '63',
              },
              {
                adjustedCount: 2950592,
                adjustedRate: 1.55,
                count: 30932336,
                fieldDuration: 4148100,
                rate: 1.08,
                version: '62',
              },
              {
                adjustedCount: 3616486,
                adjustedRate: 1.7,
                count: 112777330,
                fieldDuration: 6100800,
                rate: 1.17,
                version: '61',
              },
            ],
        },
    ]
    const componentProps = 
      {
        history: {
          length: 2,
          action: 'POP',
          location: {},
          createHref: {},
          push: {},
          replace: {},
          go: {},
          goBack: {},
          goForward: {},
          block: {},
          listen: {},
        },
        summary: {
          application: 'firefox',
          expectedMeasures: {},
          latestVersionSeen: '64.0',
          latestVersionFieldDuration: 659700,
          channel: 'release',
          platform: 'windows',
          summaryRate: '2.51',
        },
      }
      const platformCardComponentProps = {
        ...componentProps,
        measures: sampleMeasures,
      };
    const component = shallow(<PlatformCard 
      summary={platformCardComponentProps}
      />);

    expect(component).toMatchSnapshot();
  });
});
