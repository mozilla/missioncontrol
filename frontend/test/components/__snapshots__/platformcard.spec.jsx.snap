// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformCard should display PlatformCard correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlatformCard
    summary={
      Object {
        "history": Object {
          "action": "POP",
          "block": Object {},
          "createHref": Object {},
          "go": Object {},
          "goBack": Object {},
          "goForward": Object {},
          "length": 2,
          "listen": Object {},
          "location": Object {},
          "push": Object {},
          "replace": Object {},
        },
        "measures": Array [
          Object {
            "lastUpdated": "2018-12-18T07:40:00Z",
            "majorVersions": Array [
              Object {
                "adjustedCount": 3694128,
                "adjustedRate": 0.97,
                "count": 3694128,
                "fieldDuration": 663300,
                "rate": 0.97,
                "version": "64",
              },
              Object {
                "adjustedCount": 3638659,
                "adjustedRate": 1.61,
                "count": 66553946,
                "fieldDuration": 4248000,
                "rate": 1.15,
                "version": "63",
              },
              Object {
                "adjustedCount": 2747677,
                "adjustedRate": 1.44,
                "count": 30466929,
                "fieldDuration": 4148100,
                "rate": 1.06,
                "version": "62",
              },
              Object {
                "adjustedCount": 2761278,
                "adjustedRate": 1.3,
                "count": 89205500,
                "fieldDuration": 6100800,
                "rate": 0.93,
                "version": "61",
              },
            ],
            "name": "main_crashes",
            "versions": Array [
              Object {
                "adjustedCount": 3694128,
                "adjustedRate": 0.97,
                "count": 3694128,
                "fieldDuration": 663300,
                "rate": 0.97,
                "version": "64.0",
              },
              Object {
                "adjustedCount": 3694128,
                "adjustedRate": 0.97,
                "count": 3694128,
                "fieldDuration": 663300,
                "rate": 0.97,
                "version": "64",
              },
              Object {
                "adjustedCount": 3638659,
                "adjustedRate": 1.61,
                "count": 66553946,
                "fieldDuration": 4248000,
                "rate": 1.15,
                "version": "63",
              },
              Object {
                "adjustedCount": 2747677,
                "adjustedRate": 1.44,
                "count": 30466929,
                "fieldDuration": 4148100,
                "rate": 1.06,
                "version": "62",
              },
              Object {
                "adjustedCount": 2761278,
                "adjustedRate": 1.3,
                "count": 89205500,
                "fieldDuration": 6100800,
                "rate": 0.93,
                "version": "61",
              },
            ],
          },
          Object {
            "lastUpdated": "2018-12-18T07:40:00Z",
            "majorVersions": Array [
              Object {
                "adjustedCount": 5836415,
                "adjustedRate": 1,
                "count": 5836415,
                "fieldDuration": 663300,
                "rate": 1.53,
                "version": "64",
              },
              Object {
                "adjustedCount": 4163445,
                "adjustedRate": 1.84,
                "count": 68708257,
                "fieldDuration": 4248000,
                "rate": 1.19,
                "version": "63",
              },
              Object {
                "adjustedCount": 2950592,
                "adjustedRate": 1.55,
                "count": 30932336,
                "fieldDuration": 4148100,
                "rate": 1.08,
                "version": "62",
              },
              Object {
                "adjustedCount": 3616486,
                "adjustedRate": 1.7,
                "count": 112777330,
                "fieldDuration": 6100800,
                "rate": 1.17,
                "version": "61",
              },
            ],
            "name": "content_crashes",
            "versions": Array [
              Object {
                "adjustedCount": 5836415,
                "adjustedRate": 1.53,
                "count": 5836415,
                "fieldDuration": 663300,
                "rate": 1.53,
                "version": "64.0",
              },
              Object {
                "adjustedCount": 5836415,
                "adjustedRate": 1.53,
                "count": 5836415,
                "fieldDuration": 663300,
                "rate": 1.53,
                "version": "64",
              },
              Object {
                "adjustedCount": 4163445,
                "adjustedRate": 1.84,
                "count": 68708257,
                "fieldDuration": 4248000,
                "rate": 1.19,
                "version": "63",
              },
              Object {
                "adjustedCount": 2950592,
                "adjustedRate": 1.55,
                "count": 30932336,
                "fieldDuration": 4148100,
                "rate": 1.08,
                "version": "62",
              },
              Object {
                "adjustedCount": 3616486,
                "adjustedRate": 1.7,
                "count": 112777330,
                "fieldDuration": 6100800,
                "rate": 1.17,
                "version": "61",
              },
            ],
          },
        ],
        "summary": Object {
          "application": "firefox",
          "channel": "release",
          "expectedMeasures": Object {},
          "latestVersionFieldDuration": 659700,
          "latestVersionSeen": "64.0",
          "platform": "windows",
          "summaryRate": "2.51",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        className="missioncontrol-card"
        onClick={[Function]}
        tag="div"
      >
        <CardHeader
          className="alert-undefined"
          tag="div"
        >
          <center>
              
          </center>
        </CardHeader>
        <CardBody
          tag="div"
        >
          <div
            className="summary-rate"
          >
            <abbr
              title=""
            >
              N/A
            </abbr>
          </div>
          <table
            className="table table-sm summary-table"
          >
            <tbody />
          </table>
        </CardBody>
      </Card>,
      "className": "missioncontrol-card-link",
      "href": "#/undefined/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            className="alert-undefined"
            tag="div"
          >
            <center>
                
            </center>
          </CardHeader>,
          <CardBody
            tag="div"
          >
            <div
              className="summary-rate"
            >
              <abbr
                title=""
              >
                N/A
              </abbr>
            </div>
            <table
              className="table table-sm summary-table"
            >
              <tbody />
            </table>
          </CardBody>,
        ],
        "className": "missioncontrol-card",
        "onClick": [Function],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <center>
                
            </center>,
            "className": "alert-undefined",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "  ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "  ",
              undefined,
            ],
            "type": "center",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="summary-rate"
              >
                <abbr
                  title=""
                >
                  N/A
                </abbr>
              </div>,
              <table
                className="table table-sm summary-table"
              >
                <tbody />
              </table>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <abbr
                  title=""
                >
                  N/A
                </abbr>,
                "className": "summary-rate",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "N/A",
                  "title": "",
                },
                "ref": null,
                "rendered": "N/A",
                "type": "abbr",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tbody />,
                "className": "table table-sm summary-table",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
              "type": "table",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          className="missioncontrol-card"
          onClick={[Function]}
          tag="div"
        >
          <CardHeader
            className="alert-undefined"
            tag="div"
          >
            <center>
                
            </center>
          </CardHeader>
          <CardBody
            tag="div"
          >
            <div
              className="summary-rate"
            >
              <abbr
                title=""
              >
                N/A
              </abbr>
            </div>
            <table
              className="table table-sm summary-table"
            >
              <tbody />
            </table>
          </CardBody>
        </Card>,
        "className": "missioncontrol-card-link",
        "href": "#/undefined/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader
              className="alert-undefined"
              tag="div"
            >
              <center>
                  
              </center>
            </CardHeader>,
            <CardBody
              tag="div"
            >
              <div
                className="summary-rate"
              >
                <abbr
                  title=""
                >
                  N/A
                </abbr>
              </div>
              <table
                className="table table-sm summary-table"
              >
                <tbody />
              </table>
            </CardBody>,
          ],
          "className": "missioncontrol-card",
          "onClick": [Function],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <center>
                  
              </center>,
              "className": "alert-undefined",
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "  ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "  ",
                undefined,
              ],
              "type": "center",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div
                  className="summary-rate"
                >
                  <abbr
                    title=""
                  >
                    N/A
                  </abbr>
                </div>,
                <table
                  className="table table-sm summary-table"
                >
                  <tbody />
                </table>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <abbr
                    title=""
                  >
                    N/A
                  </abbr>,
                  "className": "summary-rate",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "N/A",
                    "title": "",
                  },
                  "ref": null,
                  "rendered": "N/A",
                  "type": "abbr",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tbody />,
                  "className": "table table-sm summary-table",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "tbody",
                },
                "type": "table",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
